### DISCUSSION TITLE
How to speed up calculations

### DISCUSSION THREAD

**Discussion by Superharz** [scikit-hep/awkward#810] [Q&A]
I have two large nested awkward arrays ```a``` and ```b```.
I want to calculate ```a - b```.
This operation takes about 30Gb of my RAM. However, the CPU is only at 10% and it seems like only 4 of my 16 logical cores are used.
My question: Is it possible to speed up the calculation and if yes: How?

**Comment by jpivarski** [comment]
Awkward Array does not automatically parallelize, and since most operations do not release the [Python GIL](https://realpython.com/python-gil/) (yet), a single process is not going to be able to take advantage of multiple cores.

To answer your question, I'd have to know a lot more about what `a`, `b`, and `-` are (if this is not using 100% of one CPU, they're probably not numbers, and so `-` is some [overloaded behavior](https://awkward-array.readthedocs.io/en/latest/ak.behavior.html)). However, I would guess that `a` and `b` are being lazily loaded from a file to use so much memory and not be CPU-bound.

**Reply by Superharz** [reply]
Thanks for your explanation.
I think you are right. The array I used was too big for my RAM. I found this out by flattening the the array with ```axis = None```.
I tried it again with smaller arrays ```a``` and ```b``` and flattened them to load them completly into my RAM. The later substraction ``` a - b``` happened almost instant.
